### PR TITLE
Add application insights, properties volume and update some dependencies.

### LINKED ISSUE #43: Add application insights integration

### PR DESCRIPTION
**Pull request by linusnorton**
### JIRA link ###

Fixes #43 

### Change description ###

Add application insights, properties volume and update some dependencies.

Also fix an issue with the init.sh script on linux. See [this](https://unix.stackexchange.com/questions/253892/syntax-error-unexpected-when-creating-an-array)

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app.ts [hunk 1]
```diff
@@ -19,13 +21,11 @@ const developmentMode = env === 'development';
 export const app = express();
 app.locals.ENV = env;
 
-// setup logging of HTTP requests
-app.use(Express.accessLogger());
```
Comment (linusnorton) [inline src/main/app.ts:23]: Request logging is now done by appinsights

**Comment by linusnorton** [comment]
@timja I think this is good to go but it does make some assumptions about the appinsights instance being set up and the key being available from a vault. Can you double check the chart values.yml to make sure you're happy with that?

File: charts/rpe-expressjs-template/values.yaml [hunk 2]
```diff
@@ -3,4 +3,9 @@ nodejs:
   aadIdentityName: rpe
   ingressHost: rpe-expressjs-template.{{ .Values.global.environment }}.platform.hmcts.net
   image: 'hmctspublic.azurecr.io/rpe/expressjs-template:latest'
+  keyVaults:
+    rpe:
+      resourceGroup: rpe
```
Comment (timja) [inline charts/rpe-expressjs-template/values.yaml:8]: resource group isn't needed
```suggestion
```

**Review by timja — APPROVED** [review]
looks reasonable

File: charts/rpe-expressjs-template/values.yaml [hunk 3]
```diff
@@ -3,4 +3,8 @@ nodejs:
   aadIdentityName: rpe
   ingressHost: rpe-expressjs-template.{{ .Values.global.environment }}.platform.hmcts.net
   image: 'hmctspublic.azurecr.io/rpe/expressjs-template:latest'
+  keyVaults:
```
Comment (adusumillipraveen) [inline charts/rpe-expressjs-template/values.yaml:6]: Should we comment this section by default ? and mention in readme/ comments ? I am conscious that adding it by default means they cannot get it past preview till they create the vault, appinsights.

File: src/main/modules/properties-volume/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import { Application } from 'express';
+import { get, set } from 'lodash';
+
+export class PropertiesVolume {
+
+  enableFor(server: Application): void {
+    if (server.locals.ENV !== 'development') {
+      propertiesVolume.addTo(config);
+
+      if (config.has('secrets.rpe.AppInsightsInstrumentationKey')) {
```
Comment (adusumillipraveen) [inline src/main/modules/properties-volume/index.ts:12]: Can you abstract this to a method like in https://github.com/hmcts/div-petitioner-frontend/blob/master/app/core/setup/setupSecrets.js#L4

**Review by adusumillipraveen — APPROVED** [review]
Minor comments, but looks good otherwise.

**Comment by adusumillipraveen** [comment]
Also, we will soon be migrating to use https://github.com/hmcts/template-expressjs with backstage. It is more easy to implement this there with a conditional flag around appininsights. Please see if you can copy your changes in there as well :)

**Pull request merged**